### PR TITLE
Feature/test cases/api integration

### PR DESCRIPTION
**Pull request by AsmitaMargaje**
resolved pr changes and completed download test draft integartion

### CODE REVIEW

**Comment by AsmitaMargaje** [comment]
completed UI Integration
1. Review Comment Master
2. Testing Type Master
3. Testing Group Master
4. Function Master

completed api Integration
1. Review Comment Master
2. Testing Type Master
3. Testing Group Master
4. Function Master

Completed download test draft file api integration
Remove inline css

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+import { getReviewCommentMasterListThunk } from "../../../redux/services/testCases/reviewCommentMaster";
+import { customSearchHandler } from "../../../utils/customFunction";
+import TableLoadingSkelton from "../../../components/custom/loader/TableLoadingSkelton";
+import { Col, Row } from "react-bootstrap";
+
+function ReviewCommentMasterComponent() {
+  const dispatch = useDispatch();
+
+  // // redux state
+  const { reviewCommentMasetrList, isLoading } = useSelector(
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:16]: Add spell check extention,
1.reviewCommentMasetrList
2.filteredRevieCommentMasterList

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+          }
+        />
+      ),
+      sortable: false,
+      width: "70px",
+    },
+
+    {
+      name: "Status",
+      selector: (row) => (row.is_active === 1 ? "Active" : "Deactive"),
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:82]: If active , text color should be green or else red

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+  };
+
+  // Function to handle reset button click
+  const handleReset = () => {
+    setSearchValue("");
+    setFilteredReviewCommentkMasterList(reviewCommentMasetrList);
+  };
+
+  // Update the useEffect to update the filtered list when reviewCommentMasetrList changes
+  useEffect(() => {
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:46]: Its standard practice to call useEffect at end , not randomly anywhere.

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+    {
+      name: "Updated By",
+      selector: (row) => row.updated_by,
+      sortable: false,
+      width: "175px",
+    },
+  ];
+
+  const transformDataForExport = (data) => {
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:122]: Create reusable transformDataForExport function, which can be reused multiple times

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+  useEffect(() => {
+    dispatch(getReviewCommentMasterListThunk());
+  }, []);
+
+  return (
+    <div className="container-xxl">
+      <div className="d-flex justify-content-between">
+        <PageHeader headerTitle="Review Comment Master" />
+        <div style={{ marginTop: "-30px" }}>
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:141]: Avoid inline style

File: src/screens/TestCases/ReviewCommentMaster/Validation/ReviewCommentMasterModal.js [hunk 6]
```diff
@@ -0,0 +1,174 @@
+import React, { useEffect, useState } from "react";
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/Validation/ReviewCommentMasterModal.js:1]: UseEffect not used still imported

File: src/screens/TestCases/ReviewCommentMaster/Validation/ReviewCommentMasterModal.js [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+import { Field, Form, Formik } from "formik";
+import { Col, Row, Stack, Spinner } from "react-bootstrap";
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/Validation/ReviewCommentMasterModal.js:3]: stack, spinner not used still imported

File: src/screens/TestCases/ReviewCommentMaster/Validation/ReviewCommentMasterModal.js [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+  const addEditReviewCommentInitialValue = {
+    reviewer_comment:
+      type === "EDIT" ? currentReviewCommentData?.reviewer_comment : "",
+    remark: type === "EDIT" ? currentReviewCommentData?.remark || "" : "",
+    is_active:
+      type === "EDIT" ? currentReviewCommentData?.is_active?.toString() : 1,
+  };
+
+  // // redux state
+  const { isLoading } = useSelector((state) => state?.reviewCommentMaster);
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/Validation/ReviewCommentMasterModal.js:34]: isLoading not used ,avoid rebundant usage.

File: src/screens/TestCases/ReviewCommentMaster/Validation/AddReviewComment.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import * as Yup from "yup";
+import {
+  ALPHA_NUMERIC_REGEX,
+  ONLY_CHARACTER_REGEX,
+} from "../../../../settings/constants";
+export const addReviewCommentValidation = Yup.object().shape({
+  reviewer_comment: Yup.string()
+    .max(50, "Review comment title must be at most 50 characters")
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/Validation/AddReviewComment.js:8]: It is mandatory field , but also its getting submitedd even if kept empty

File: src/screens/TestCases/ReviewCommentMaster/Validation/AddReviewComment.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    .matches(ONLY_CHARACTER_REGEX, "Review comment title must be alphabets"),
+  remark: Yup.string()
+    .min(2, "Remark description must be at least 2 characters")
```
Comment (AmitSolnke) [inline src/screens/TestCases/ReviewCommentMaster/Validation/AddReviewComment.js:11]: what is the reason behind this line of code?

File: src/screens/TestCases/TestDraft/DownloadFormatFileModal.js [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+    const { project_id, module_id, submodule_id } = formData.formData;
+
+    dispatch(
+      downloadFormatFileThunk({
+        project_id,
+        module_id,
+        submodule_id,
+
+        onSuccessHandler: () => {
+          // setOpenConfirmModal({ open: false });
```
Comment (AmitSolnke) [inline src/screens/TestCases/TestDraft/DownloadFormatFileModal.js:81]: why there is commented code?

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js (hunk 1 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:16]: Added spell check extention, and updated changes

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js (hunk 2 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:82]: badge added

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js (hunk 5 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:141]: removed inline style

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js (hunk 3 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:46]: updated the changes

File: src/screens/TestCases/ReviewCommentMaster/Validation/ReviewCommentMasterModal.js (hunk 6 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/ReviewCommentMaster/Validation/ReviewCommentMasterModal.js:1]: removed

File: src/screens/TestCases/ReviewCommentMaster/Validation/AddReviewComment.js (hunk 10 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/ReviewCommentMaster/Validation/AddReviewComment.js:11]: to check the length of reviewer comment

File: src/screens/TestCases/TestDraft/DownloadFormatFileModal.js (hunk 11 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/TestDraft/DownloadFormatFileModal.js:81]: removed

**Pull request merged**